apollo-federation/src/merge.rs

```diff
@@ -600,23 +674,81 @@ impl Merger {
     }
 
     fn merge_scalar_type(
-        &self,
+        &mut self,
         types: &mut IndexMap<Name, ExtendedType>,
+        metadata: &Option<&LinksMetadata>,
         subgraph_name: Name,
         scalar_name: NamedType,
         ty: &Node<ScalarType>,
     ) {
         let existing_type = types
             .entry(scalar_name.clone())
-            .or_insert(copy_scalar_type(scalar_name, ty));
+            .or_insert(copy_scalar_type(scalar_name.clone(), ty));
+
         if let ExtendedType::Scalar(s) = existing_type {
             let join_type_directives =
                 join_type_applied_directive(subgraph_name.clone(), iter::empty(), false);
             s.make_mut().directives.extend(join_type_directives);
+            self.add_inaccessible(metadata, &mut s.make_mut().directives, &ty.directives);
         } else {
             // conflict?
         }
     }
+
+    fn add_inaccessible(
+        &mut self,
+        metadata: &Option<&LinksMetadata>,
+        new_directives: &mut apollo_compiler::schema::DirectiveList,
+        original_directives: &apollo_compiler::schema::DirectiveList,
+    ) {
+        let federation_identity =
+            metadata.and_then(|m| m.by_identity.get(&Identity::federation_identity()));
+
+        let inaccessible_directive_name = federation_identity
```

Comment:
I wonder if we should calculate this once, then pass it down instead of metadata?